metrotranscom/doorway
# fix: frame ancestors approach

# Pull Request Template

## Issue Overview

This PR addresses #448 


## Description

This PR adds a Content Security Policy to the public and partners next config. I've found some simple online tools to check if individual pages can be framed but they do not work with localhost. Given this, the resource below backing this approach, and that the headers seem to be set correctly on localhost. I feel comfortable mergining this in and testing the fame-ability of staging.

Source in issue describing x-frame options as common defense: https://blog.qualys.com/vulnerabilities-threat-research/2015/10/20/clickjacking-a-common-implementation-mistake-that-can-put-your-websites-in-danger 

Source describing which x-frame option is best for our use: https://cheatsheetseries.owasp.org/cheatsheets/Clickjacking_Defense_Cheat_Sheet.html


Related Nextjs Docs: https://nextjs.org/docs/app/building-your-application/configuring/content-security-policy#without-nonces

## How Can This Be Tested/Reviewed?

This can be seen in the in the network tab when running the partners or public site. Click on index.js or app.js in the network tab under the response headers to see Content-Security-Policy: frame-ancestors none;

<img width="743" alt="Screenshot 2024-04-22 at 4 26 04 PM" src="https://github.com/metrotranscom/doorway/assets/53269332/dde98518-5441-4b6d-b7ba-ada8136545d0">

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
